tests/python_on_whales/test_client_config.py

```diff
@@ -36,3 +37,53 @@ def test_pretty_exception_message_and_report(mocker):
         raise IndexError
 
     assert Path(word).read_text() == json.dumps(fake_json_message, indent=2)
+
+
+def test_compose_env_file():
+    """Test that the deprecated `compose_env_file` gives a warning, and adds the `--env-file` argument to the compose command"""
+    with pytest.warns(UserWarning):
+        example_env_file_name = "example.env"
+        client_config = ClientConfig(compose_env_file=example_env_file_name)
+        assert client_config.docker_compose_cmd.count("--env-file") == 1
+        # Since we are operating of a list of commands, we first find the index of the `--env-file` argument, then we check that the next argument is the file name.
+        index = client_config.docker_compose_cmd.index("--env-file")
+        assert client_config.docker_compose_cmd[index + 1] == example_env_file_name
+
+
+@pytest.mark.parametrize(
+    "compose_env_files",
+    [
+        ["example1.env", "example2.env"],
+        ["example1.env"],
+    ],
+    ids=["multiple_files", "single_file"],
+)
+def test_compose_env_files(compose_env_files: Sequence[str]):
+    """Test that using only `compose_env_files` adds all the files to the command line arguments"""
+    client_config = ClientConfig(compose_env_files=compose_env_files)
+    # Since we are operating of a list of commands, we first find the indices of the `--env-file` argument, then we check that the next argument is the file name.
+    indices = [
+        index
+        for index, item in enumerate(client_config.docker_compose_cmd)
+        if item == "--env-file"
+    ]
+    assert len(indices) == len(compose_env_files)
+    for index, env_file in zip(indices, compose_env_files):
+        assert client_config.docker_compose_cmd[index + 1] == env_file
+
+
+def test_compose_env_files_and_env_file():
```

Comment:
The tests are good, keep them :) I'd also like a test which actually runs the docker compose command and checks that the env files are loaded correctly. For example with `docker.compose.config()`. If for example, you made a typo in `--env-file` in the unit test and the main implementation, your tests can't currently detect it. Running the true command in test would detect such an error.

Reply:
Added some more tests now that actually run the command by using the equivalent of `docker compose run`. Some temporary env-files are made, and the containers are then inspected to verify that the actual environment variables set in the container are as expected. Pretty verbose, and I am sure it can be done simpler, but let me know if this is what you wanted :)

Reply:
Aha, yeah that is a lot simpler. Updated now :)